### ISSUE TITLE
Organization plan

### ISSUE THREAD

**Issue by kepta** [bangle-io/bangle.dev#219]
TLDR: break up `/core` into `/editor` and `/core-components`.

I feel the repository has grown quite a bit in terms of code size and it is worth refactoring the code in more appropriate `npm` packages. 

I am imagine having a structure like:

- tooling/ (internal things)
   - integration-tests
   - scripts
- lib/
    - @bangle.dev/editor
    - @bangle.dev/react

    - @bangle.dev/markdown
    - @bangle.dev/utils
    - @bangle.dev/collab-client
    - @bangle.dev/collab-server
    - @bangle.dev/disk
    - @bangle.dev/pm
- components/
    - @bangle.dev/react-menu
    - @bangle.dev/core-components
    - @bangle.dev/table
    - @bangle.dev/emoji
    - @bangle.dev/wiki-link
    - @bangle.dev/trailing-node
    - @bangle.dev/timestamp
    - @bangle.dev/text-formatting
    - @bangle.dev/search
    - @bangle.dev/react-stopwatch
    - @bangle.dev/markdown-front-matter
    - @bangle.dev/react-emoji-suggest
    - @bangle.dev/react-sticker

**Comment by lucywang000** [comment]
>  break up /core into /editor and /core-components.

which part of `/core` would go into each of the two new packages respectively?

**Comment by kepta** [comment]
> which part of /core would go into each of the two new packages respectively?

I am thinking of moving all code in [components](https://github.com/bangle-io/bangle.dev/tree/master/core/components) to `core-components` and the remaining code that sets up editor will stay in editor.
